src/content/learn/typescript.md

```diff
@@ -119,54 +118,54 @@ export default App = AppTSX;
+As definições de tipos em `@types/react` incluem tipos para os hooks nativos, então você pode usá-los em seus componentes sem nenhuma configuração adicional. Eles são construídos levando em conta o código que você escreve em seu componente, então você terá a [inferência de tipo](https://www.typescriptlang.org/pt/docs/handbook/type-inference.html) na maior parte do tempo e idealmente você não precisará se preocupar com as minúcias de fornecer tipos.
 
-However, we can look at a few examples of how to provide types for hooks.
+Entretanto, podemos observar alguns exemplos de como fornecer tipos para hooks.
 
 ### `useState` {/*typing-usestate*/}
 
-The [`useState` hook](/reference/react/useState) will re-use the value passed in as the initial state to determine what the type of the value should be. For example:
+O [hook `useState`](/reference/react/useState) irá reutilizar o valor passado como state inicial para determinar qual o tipo o valor deve ser. Por exemplo: 
 
 ```ts
-// Infer the type as "boolean"
+// Tipo inferido como "boolean"
 const [enabled, setEnabled] = useState(false);
 ```
 
-Will assign the type of `boolean` to `enabled`, and `setEnabled` will be a function accepting either a `boolean` argument, or a function that returns a `boolean`. If you want to explicitly provide a type for the state, you can do so by providing a type argument to the `useState` call:
+Atribuirá o tipo `boolean` à `enabled` e `setEnable` será uma função que ou aceita um `boolean` como argumento ou uma função que retorna `boolean`. Se você deseja informar explicitamente um tipo para o state, você pode informat um tipo como argumento à chamada do `useState`:
 
 ```ts 
-// Explicitly set the type to "boolean"
+// Explicitamente define o tipo como "boolean"
 const [enabled, setEnabled] = useState<boolean>(false);
 ```
 
-This isn't very useful in this case, but a common case where you may want to provide a type is when you have a union type. For example, `status` here can be one of a few different strings:
+Não é algo muito útil neste caso, mas um caso comum seria onde você deseja informar um tipo que representa um _union type_. Por exemplo, `status` aqui pode ser uma dentre algumas _strings_:
 
 ```ts
-type Status = "idle" | "loading" | "success" | "error";
+type Status = "ocioso" | "carregando" | "sucesso" | "erro";
 
-const [status, setStatus] = useState<Status>("idle");
+const [status, setStatus] = useState<Status>("ocioso");
 ```
 
-Or, as recommended in [Principles for structuring state](/learn/choosing-the-state-structure#principles-for-structuring-state), you can group related state as an object and describe the different possibilities via object types:
+Ou, como recomendado em [Principles for structuring state](/learn/choosing-the-state-structure#principles-for-structuring-state), você pode agrupar estados relacionados em um objeto descrevendo suas diferentes possibilidades através de tipos objetos:
```

Comment:
Podes traduzir essa parte tambem? [Principles for structuring state]

Reply:
Fiquei na dúvida sobre traduzir ou não a descrição do link já que a página original está em inglês. Mas faz sentido traduzir, agradeço as observações.